### ISSUE TITLE
strange comments in ruleset

### ISSUE THREAD

**Issue by cybermcm** [pitkley/dfw#134]
I set a few initialization rules to fit my needs:
```
[initialization]
rules = [
"flush table inet filter",
"delete chain inet filter input",
"delete chain inet filter forward",
"delete chain inet filter output",
"add chain inet filter input { type filter hook input priority 0 ; policy drop ; }",
"add chain inet filter forward { type filter hook forward priority 0 ; policy drop ; }",
"add chain inet filter output { type filter hook output priority 0 ; policy accept ; }",
"add rule inet filter input ct state established, related accept",
"add rule inet filter input ct state invalid drop",
"add rule inet filter input iifname lo accept",
"add rule inet filter input icmp type echo-request accept",
"add rule inet filter input tcp dport 22 ct state new,established accept"
]
```
It seems to work but nft list ruleset shows some "strange" comments:
```
table inet filter {
        chain input {
                type filter hook input priority 0; policy drop;
                ct state invalid drop comment "DFW-MARKER:defaults;filter;input;ct-state-invalid-drop"
                ct state { established, related } accept comment "DFW-MARKER:defaults;filter;input;ct-state-relatedestablished-accept"
                mark & 0x000000df == 0x000000df accept comment "DFW-MARKER:defaults;filter;input;meta-mark"
                ct state established,related accept
                ct state invalid drop
                iifname "lo" accept
                icmp type echo-request accept
                tcp dport ssh ct state established,new accept
        }

        chain forward {
                type filter hook forward priority 0; policy drop;
                ct state invalid drop comment "DFW-MARKER:defaults;filter;forward;ct-state-invalid-drop"
                ct state { established, related } accept comment "DFW-MARKER:defaults;filter;forward;ct-state-relatedestablished-accept"
                mark & 0x000000df == 0x000000df accept comment "DFW-MARKER:defaults;filter;forward;meta-mark"
        }

        chain output {
                type filter hook output priority 0; policy accept;
        }
}
```
input chain
```
ct state invalid drop comment "DFW-MARKER:defaults;filter;input;ct-state-invalid-drop"
ct state { established, related } accept comment "DFW-MARKER:defaults;filter;input;ct-state-relatedestablished-accept"
mark & 0x000000df == 0x000000df accept comment "DFW-MARKER:defaults;filter;input;meta-mark"
```
forward chain
```
ct state invalid drop comment "DFW-MARKER:defaults;filter;forward;ct-state-invalid-drop"
ct state { established, related } accept comment "DFW-MARKER:defaults;filter;forward;ct-state-relatedestablished-accept"
mark & 0x000000df == 0x000000df accept comment "DFW-MARKER:defaults;filter;forward;meta-mark"
```
is there something wrong with my rules?

**Comment by pitkley** [comment]
There are no issues, in fact this is perfectly fine and how it should be. [As described in the migration documentation](https://github.com/pitkley/dfw/blob/fa4c44a112478cabdfb76de5b98fa93ee0196aac/MIGRATION-v0.x-to-v1.0.md#new-keys), DFW needs a way to hook into pre-existing tables to be able to create a definitive verdict (drop/reject/accept decision) for a packet.

With iptables we got this "for free": everything was in one table. With DFW I decided that I want to use the multiple table feature, which requires special care to actually allow traffic.

The way nftables works is that if you have multiple chains, a packet is only approved if _all applicable chains_ approve the packet in order (lowest priority-value to highest priority-value). If any single one chain drops or rejects a packet, the following chains will not be evaluated.

Since DFW creates its chains with a priority lower than default, it is able to block packets before they arrive at the users table, but it is not able to approve it unless the users table also approves it. This is what happens with the `mark` rules: they check if DFW has previously approved a packet and approves it again.

The comments are used for DFW to know if the necessary rules exist or if they have to be created (to not unnecessarily duplicate already existing rules).

Does this help? :slightly_smiling_face:
